---
title: "Cystovirus Genome Sequencing"
output: html_notebook
---

#Cystovirus Genome Sequecing

## Data 

### Illumina Sequencing Runs

#### Run AW070319
Illumina PE 150, MiSeq library 'AW070319' run at the UCD GC. 12.3M reads pF (of which 9.6% are PhiX) with an overall Q30>89.4%. Data received by email on July 10, 2019 at 10:09am  

Run details: 
https://slims.bioinformatics.ucdavis.edu/solexa/view_run.php?id=3673
Demultiplexed fastq files are here: http://slimsdata.genomecenter.ucdavis.edu/Data/uod8hs5jdq/190708_M02034_0427_MS8249685-300V2/MiSeqAnalysis/

Now downloading to Crick Cluster

```{bash}
ssh sldmunoz@crick.cse.ucdavis.edu

cd /group/diazmunozlab/data/

mkdir cysto_genome_sequences/

cd /group/diazmunozlab/data/cysto_genome_sequences/

mkdir AW070319

cd AW070319

#Download data from GC Server
wget -r -nH -nc -np -R index.html*    “http://slimsdata.genomecenter.ucdavis.edu/Data/uod8hs5jdq/190708_M02034_0427_MS8249685-300V2/"

#Actually run as a job
sbatch -c1 --mem=1g -t05:00:00 --wrap="wget -r -nH -nc -np -R index.html* http://slimsdata.genomecenter.ucdavis.edu/Data/uod8hs5jdq/190708_M02034_0427_MS8249685-300V2/"

#Check on progress and successful upload
tail slurm-976115.out
#FINISHED --2021-04-06 12:58:34--
#Total wall clock time: 3m 27s
#Downloaded: 697 files, 5.1G in 2m 42s (32.6 MB/s)
```


#### Run AW020720
Illumina PE 75, MiSeq sample 'AW020720' run at the UCD GC. 14.8 M reads passing filter (of which 21% are PhiX), and an overall Q30 > 96%. Data received via email February 20, 2020 at 1:47:58

Details on your run can be found at this link: https://slims.bioinformatics.ucdavis.edu/solexa/view_run.php?id=3996
Fastq files (gzipped) are here: http://slimsdata.genomecenter.ucdavis.edu/Data/7h6qb6smm4/200219_M00384_0230_MS8565112-150V3/

FASTQ's are in "Data/7h6qb6smm4/200219_M00384_0230_MS8565112-150V3/MiSeqAnalysis/"

Ignore sample labels for now.

```{bash}
ssh sldmunoz@crick.cse.ucdavis.edu

cd /group/diazmunozlab/data/

mkdir cysto_genome_sequences/

cd /group/diazmunozlab/data/cysto_genome_sequences/

#Download data from GC Server
wget -r -nH -nc -np -R index.html*    “http://slimsdata.genomecenter.ucdavis.edu/Data/7h6qb6smm4/200219_M00384_0230_MS8565112-150V3/"

#Actually run as a job
sbatch -c1 --mem=1g -t05:00:00 --wrap="wget -r -nH -nc -np -R index.html* http://slimsdata.genomecenter.ucdavis.edu/Data/7h6qb6smm4/200219_M00384_0230_MS8565112-150V3/"

#Check on progress and successful upload
tail slurm-833314.out
#FINISHED --2021-02-12 11:26:14--
#Total wall clock time: 6m 1s
#Downloaded: 28256 files, 4.5G in 55s (82.7 MB/s)
```




### Independent Runs from NYU Genome Center
(to be used later)

## Analysis of Lab Generated Sequence Data  

### Proposed pipeline:

1. Read QC, potential adapter and barcode removal
2. Potential removal of bacterial host reads: evaluate whether necessary 
3. Genome assembly with spades with --meta option or the new MetaviralSPAdes  
4. Filter generated contigs by size (see segments below), quality, and closely related matches

## Reference information

## Phage
phi-6 is the cannonical member of the Cystoviridae:
https://viralzone.expasy.org/165

It has a tri-segmented genome, with a small, medium, and large segment (L=6374 nt; M=4063 nt, S = 2948 nt):
https://www.ncbi.nlm.nih.gov/genomes/GenomesGroup.cgi?taxid=10879


## Host
The host used to isolate and propagate all these cystoviruses is Pseudomonas syringae pv. phaseolicola strain HB10Y (ATCC)

Genome is available here:
https://www.ncbi.nlm.nih.gov/nuccore/NZ_LGKX01000037.1

## Discovery of Cystoviruses in Metagenomes
If the analysis of our sequence generated in house is not enough, searching metagenomes (random hexamer primed RNAseq is prob. best best). Feel free to employ any tools you have knowledge of.

Paper that did metagenomic search that yielded Cystoviruses:
https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1002409

Note plant, water, and soil RNAseq data sets prob. are best bet. 